### ISSUE TITLE
Highly auditble and reproducible build environment for openenclave using NIX should be added

### ISSUE THREAD

**Issue by yakman2020** [openenclave/openenclave#3729]
NIX is a package manager/build system which incorporates the principles of transparency and reproducibility at a very basic level. The result is that it is possible to provable trace the resulting .deb file through the dependent libraries and tools to the end result. This greatly increases the level of confidence a client project can have in the resulting package.

I propose to wrap the openenclave  build process in a nix expression, using a docker container specified by sha, to build the standard openenclave project in an isolated environment. A single command line argument would be enough to  cause the project and all of its dependencies to be built from known sources, or to build openenclave with cached dependencies in pretty much the usual time. 

NIX is an excellent system but it has some limitations.  I am not proposing that NIX become the mainline build process for production. 

- NIX supports linux only, and is unlikely to ever support Windows.
- IMO NIX package repos are not at industrial levels of quality and reliability.  To use the build in production would require someone to manage a curated package repo. The design of NIX facilitates that, but it is work that would be required for production.
-  Nix does support aarch64, but my submission would not be vetted on that platform.

With agreement in principle I can submit a detailed design doc.

**Comment by CodeMonkeyLeet** [comment]
@yakman2020 Yes, please submit a design doc for this and also submit an additional issue to track how this proposal would be integrated with CI/CD. Does not have to be in v0.13, adding this to the v0.14 milestone which can be revisited in the next sprint.

@radhikaj FYI for SIG-Release tracking.
